File: DSA/Sarthak_Jain_590014936/DAY_10/Day_10_Question_1.py
#Sarthak Jain 590014936
#Day 10 Question 1
#String Comparison

s1 = input("Enter first string: ")
s2 = input("Enter second string: ")

# Exact equality
if s1 == s2:
    print("Equal")
else:
    print("Not equal")

# Case-insensitive equality
if s1.lower() == s2.lower():
    print("Equal ignoring case")
else:
    print("Not equal ignoring case")

# Alphabetical (lexicographical whaatt!) comparison
if s1 == s2:
    print("Strings are identical")
elif s1 < s2:
    print('"' + s1 + '" comes before "' + s2 + '"')
else:
    print('"' + s2 + '" comes before "' + s1 + '"')

File: DSA/GEHNA_KHANNA_590013140/day10question1.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main() {
    char str1[100], str2[100];
    int i;

    printf("Enter String 1: ");
    fgets(str1, sizeof(str1), stdin);
    printf("Enter String 2: ");
    fgets(str2, sizeof(str2), stdin);

    for (i = 0; str1[i] != '\0'; i++) 
    {
        if (str1[i] == '\n') {
            str1[i] = '\0';
            break;
        }
    }

    for (i = 0; str2[i] != '\0'; i++) {
        if (str2[i] == '\n') {
            str2[i] = '\0';
            break;
        }
    }

    if (strcmp(str1, str2) == 0)
        printf("Equal, ");
    else
        printf("Not equal, ");

    i = 0;
    while (str1[i] && str2[i]) {
        if (tolower(str1[i]) != tolower(str2[i]))
            break;
        i++;
    }

    if (str1[i] == '\0' && str2[i] == '\0')
        printf("Equal ignoring case, ");
    else
        printf("Not equal ignoring case, ");

    if (strcmp(str1, str2) < 0)
        printf("\"%s\" comes before \"%s\"\n", str1, str2);
    else if (strcmp(str1, str2) > 0)
        printf("\"%s\" comes after \"%s\"\n", str1, str2);
    else
        printf("Strings are identical\n");

    return 0;
}
